### ISSUE TITLE
[New rule] avoid-use-expanded-as-spacer

### ISSUE THREAD

**Issue by furaiev** [dart-code-checker/dart-code-metrics#898]
**Please describe what the rule should do:** the rule should detect `Expanded` widgets that contain empty `SizedBox/Container` and propose to replace them with `Spacer()`

**If your rule is inspired by other please provide link to it:** -

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

```dart
// Bad

Column(
  children: [
    const Expanded(child: SizedBox()),
    ...
  ]
)

// Good

Column(
  children: [
    const Spacer(),
    ...
  ]
)


```

**Comment by incendial** [comment]
Available in 4.17.0 🚀
